src/components/MovieSlider.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  Box,
+  Flex,
+  Heading,
+  Stack,
+  Tab,
+  TabList,
+  Tabs,
+} from "@chakra-ui/react";
+import MovieCard from "@/components/MovieCard";
+import React from "react";
+
+interface TabsOptions {
+  key: string;
+  title: string;
+}
+
+interface HeadingMovieSliderProps {
+  title: string;
+  tabOptions: TabsOptions[];
+  onTabChange: (index: number) => void;
+}
+
+const HeadingMovieSlider = ({
+  title,
+  tabOptions,
+  onTabChange,
+}: HeadingMovieSliderProps) => {
+  return (
+    <Stack
+      p={6}
+      flexDirection={["column", "row"]}
+      alignItems={["start", "center"]}
+      spacing={[4, 0]}
+    >
+      <Heading as={"h6"} fontSize={"2xl"} mr={4}>
+        {title}
+      </Heading>
+      <Tabs
+        variant="soft-rounded"
+        colorScheme="blue"
+        borderRadius={"full"}
+        border={"1px"}
+        height={"8"}
+        onChange={(index) => onTabChange(index)}
+      >
+        <TabList height={"full"}>
+          {tabOptions.map((option) => (
+            <Tab key={option.key} value={option.key} color={"gray.800"}>
+              {option.title}
+            </Tab>
+          ))}
+        </TabList>
+      </Tabs>
+    </Stack>
+  );
+};
+
+interface Movie {
+  id: number;
+  posterPath: string;
+  title: string;
+  releaseDate: string;
+  voteCount: number;
+}
+
+type MovieSliderProps =
+  | { state: "loading"; tabOptions: TabsOptions[]; title: string }
+  | {
+      state: "error";
+      message: string;
+      tabOptions: TabsOptions[];
+      title: string;
+    }
+  | {
+      state: "loaded";
+      movies: Record<string, Movie[]>;
+      tabOptions: TabsOptions[];
+      title: string;
+    };
```

Comment:
`title` dan `tabOptions` ini kan selalu ada di semua state. Jadi bisa dibuat gini : 

```suggestion

type MovieSliderProps =
 { title: string; tabOptions: TabOption[]; } &
 (
    | { state: "loading"; }
    | { state: "error";  message: string; }
    | { state: "loaded"; movies: Record<string, Movie[]>; };
  )
```

Reply:
Oiya yaa, jadi `title` dan `tabOptions` ada di semua kondisi

Reply:
yoi